presto-main/src/main/java/com/facebook/presto/metadata/FunctionRegistry.java

```diff
@@ -467,12 +468,27 @@ public Signature resolveFunction(QualifiedName name, List<TypeSignature> paramet
             }
             Signature signature = bindSignature(function.getSignature(), boundParameters, resolvedTypes.size());
             if (signature != null) {
-                // TODO: This should also check for ambiguities
-                match = signature;
-                break;
+                coercedMatches.add(signature);
+            }
+        }
+
+        // TODO: remove when we move to a lattice-based type coercion system
+        if (coercedMatches.size() == 2) {
+            for (int i = 0; i < resolvedTypes.size(); i++) {
+                if (AMBIGUOUS_INTEGRAL_COERCION_SOURCES.contains(resolvedTypes.get(i)) &&
+                        typeManager.getType(coercedMatches.get(0).getArgumentTypes().get(i)).equals(DOUBLE) &&
```

Comment:
Why do you need to call `getType` here? I think you can just replace it with `coercredMatches.get(0).getArgumentTypes().get(i).getBase().equals(StandardTypes.DOUBLE)`